src/v/cluster/tm_stm.h

```diff
@@ -71,6 +71,23 @@ class tm_stm final : public persisted_stm {
       ss::sharded<features::feature_table>&,
       ss::sharded<cluster::tm_stm_cache>&);
 
+    void try_rm_lock(kafka::transactional_id tid) {
```

Comment:
>when we were removing a tx from the cache we could remove a mutex
even if another fiber was waiting on it and it was causing a seg
fault

the commit message makes it sound like a segfault is being fixed. but the commit itself also looks like it is adding functionality that the commit message implies was already present (removing a tx). so maybe some clarification there would be useful.